src/toolkit/matl_buy_policy.cc

```diff
@@ -136,7 +163,16 @@ std::set<RequestPortfolio<Material>::Ptr> MatlBuyPolicy::GetMatlRequests() {
   rsrc_commods_.clear();
   std::set<RequestPortfolio<Material>::Ptr> ports;
   bool make_req = buf_->quantity() < req_when_under_ * buf_->capacity();
-  double amt = TotalQty();
+  double amt;
+  
+  if (manager()->context()->time() % (active_ + dormant_) < active_) {
```

Comment:
Wonder if there is a slight performance advantage to test whether `dormant_` is greater than 0 before testing the rest of the conditions?
```suggestion
  if (dormant_ > 0 && manager()->context()->time() % (active_ + dormant_) < active_) {
```